From f4016c7323f1c07b95031edf14454b845ea68db5 Mon Sep 17 00:00:00 2001
From: Yangbo Lu <yangbo.lu@nxp.com>
Date: Fri, 14 Sep 2018 16:27:36 +0800
Subject: [PATCH 1/7] mmc: sdhci-of-esdhc: fix clock division erratum
 workaround

The workaround should be applied when setting HS400 clock.
The timing checking is not enough, and HS400 clock value
must be checked to make sure driver is setting HS400 clock.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>
[Xulin: Original patch taken from NXP lx2160a-early-access-bsp0.4-p1.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/mmc/host/sdhci-of-esdhc.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/host/sdhci-of-esdhc.c b/drivers/mmc/host/sdhci-of-esdhc.c
index 97eb9fe..67a77d06 100644
--- a/drivers/mmc/host/sdhci-of-esdhc.c
+++ b/drivers/mmc/host/sdhci-of-esdhc.c
@@ -522,7 +522,8 @@ static void esdhc_of_set_clock(struct sdhci_host *host, unsigned int clock)
 		div++;
 
 	if (esdhc->quirk_limited_clk_division &&
-		host->mmc->ios.timing == MMC_TIMING_MMC_HS400) {
+	    host->mmc->ios.timing == MMC_TIMING_MMC_HS400 &&
+	    clock == MMC_HS200_MAX_DTR) {
 		division = pre_div * div;
 		if (division <= 4) {
 			pre_div = 4;
-- 
1.7.9.5

